Forta-Bot-Challenge-4/COMP01/src/utils.ts

```diff
@@ -0,0 +1,47 @@
+import { LRUCache } from "lru-cache";
+import { THRESHOLD } from "./constants";
+import { BigNumber } from "@ethersproject/bignumber";
+export const amountCache: LRUCache<string, { timestamp: number; amount: any }> = new LRUCache({
+  max: 1000,
+});
+
+export const isOverThreshold = async (poolAddress: string, amount: any, timestamp: number): Promise<number> => {
+  const bigAmount = BigNumber.from(amount);
+  const bigThreshold = BigNumber.from(THRESHOLD);
+
+  // Check if there is an entry in the cache for the current poolAddress
+  if (amountCache.has(poolAddress)) {
+    const cachedData = amountCache.get(poolAddress) || { timestamp: 0, amount: "0" };
+    const totalAmountWithinDay = BigNumber.from(cachedData.amount).add(bigAmount);
+
+    // Update the cache with the provided timestamp and accumulated amount
+    amountCache.set(poolAddress, {
+      timestamp: timestamp,
+      amount: totalAmountWithinDay.toString(),
+    });
+
+    // Check if the total amount within the last day is over the threshold
+    if (totalAmountWithinDay.gt(bigThreshold)) {
+      return totalAmountWithinDay.sub(bigThreshold).toNumber();
```

Comment:
Currently, the amount supplied is not "reset" at any point, so once that threshold is passed, the bot will emit alerts after _every_ supply event because it continue to be above the threshold.

The amount that you are tracking over time therefore needs to be reset after it passes the threshold, that's if that _is_ your intention.